File: Data Structures & Algorithms/binary-tree-preorder-traversal/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        if(root is None):
            return ans        
        st = [root]
        while st:
            curr = st.pop()
            ans.append(curr.val)
            if(curr.right is not None):
                st.append(curr.right)
            if(curr.left is not None):
                st.append(curr.left)
        return ans
        

        

File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int maxDepth(TreeNode root) {
        if(root==null) return 0;
        return dfs(root);
    }
    public int dfs(TreeNode root){
        if(root == null){
            return 0;
        }
        int left_c = dfs(root.left);
        int right_c = dfs(root.right);
        return 1+Math.max(left_c,right_c);
    }
}
